File: apps/admin-dashboard/src/features/settings/api/supportApi.ts
import { httpClient } from '../../../shared/lib/httpClient';

export interface SupportRequestInput {
  email: string;
  subject: string;
  message: string;
  familyId?: string;
  userId?: string;
}

/** Calls the real, already-built public `/support` endpoint
 * (SupportService) — same one the Parent App uses.
 *
 * HONEST LIMITATION: this Dashboard's auth store doesn't currently
 * expose `familyId` directly, so `familyId`/`userId` are omitted
 * here — `isPriority` (Sprint 1 audit's priority_support wiring)
 * will always resolve to false for requests submitted from here,
 * unlike the Parent App where it's already wired correctly. Low
 * impact (affects only internal support-queue sorting, not access to
 * anything), not fixed here to avoid an extra API round-trip just
 * for this. */
export const supportApi = {
  submitRequest(input: SupportRequestInput) {
    return httpClient('/support', { method: 'POST', body: input, skipAuth: true });
  },
};

File: apps/parent-app/lib/features/support/api/support_api.dart
import '../../../core/network/api_client.dart';

/// Sprint 6 (Support) — calls the real, just-built `/support` endpoint
/// (public, no auth required). `skipAuth` isn't set explicitly here
/// since ApiClient attaches a bearer token automatically when a
/// session exists anyway (the backend simply ignores it for this
/// public endpoint) — no special-casing needed for the logged-in case.
class SupportApi {
  SupportApi(this._client);

  final ApiClient _client;

  Future<void> submitRequest({
    required String email,
    required String subject,
    required String message,
    String? familyId,
    String? userId,
  }) {
    return _client.post('/support', data: {
      'email': email,
      'subject': subject,
      'message': message,
      if (familyId != null) 'familyId': familyId,
      if (userId != null) 'userId': userId,
    });
  }
}
